### PR TITLE
support jdk11 

### LINKED ISSUE #503: cn.edu.tsinghua.iotdb.queryV2.engine.control.OverflowFileStreamManager uses the deprecated internal SUN API which is not available in Java 11 anymore
cn.edu.tsinghua.iotdb.queryV2.engine.control.OverflowFileStreamManager uses the deprecated internal SUN API which is not available in Java 11 anymore
try to replace the related API

### LINKED ISSUE #540: iotdb sub module unspport in jdk9
**Describe the bug**
In ut, we used `import sun.nio.ch.DirectBuffer;` it is no longer public in jdk9
`
**To Reproduce**
In JDK9 env

1. `mvn clean test -pl iotdb -am -Dtsfile.test.skip=true -Djdbc.test.skip=true -Dit.test.excludes="**/IoTDB*Test.java"`

```
[INFO] Running cn.edu.tsinghua.iotdb.queryV2.component.SegmentInputStreamWithMMapTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.021 s <<< FAILURE! - in cn.edu.tsinghua.iotdb.queryV2.component.SegmentInputStreamWithMMapTest
[ERROR] testWithMMap(cn.edu.tsinghua.iotdb.queryV2.component.SegmentInputStreamWithMMapTest)  Time elapsed: 0.021 s  <<< ERROR!
java.lang.IllegalAccessError: class cn.edu.tsinghua.iotdb.queryV2.component.SegmentInputStreamWithMMapTest (in unnamed module @0x748ef48e) cannot access class jdk.internal.ref.Cleaner (in module java.base) because module java.base does not export jdk.internal.ref to unnamed module @0x748ef48e
	at cn.edu.tsinghua.iotdb.queryV2.component.SegmentInputStreamWithMMapTest.testWithMMap(SegmentInputStreamWithMMapTest.java:52)
```

### LINKED ISSUE #552: upgrade thrift to support jdk11

### PR DESCRIPTION
**Pull request by MyXOF**
fix #552 
fix #540 
fix #503

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,16 +3,24 @@ dist: trusty
...
+#  - oraclejdk9
+#  - openjdk8
 
 before_install: true
 
 install: true
 
 script: 
+  - mvn clean package -Dmaven.test.skip=true
   - mvn clean test -pl jdbc -am -Dtsfile.test.skip=true
```
[the remark this answers is not in the document]
Comment (MyXOF) [reply .travis.yml]: I have added more comments

[the remark this answers is not in the document]
Comment (jixuan1989) [reply .travis.yml]: # For each test, travis-ci limits its output log. So we cannot test some sub-modules like tsfile, iotdb.
# We only test jdbc to ensure maven commands work correctly on travis-ci.

it does not make sense

travis is for continuous integration, but you only test JDBC?

[the remark this answers is not in the document]
Comment (MyXOF) [reply .travis.yml]: I know it is non-sense, we rely t on jenkins, you can ignore travis-ci for now. I just want to learn travis-ci script grammar

[the remark this answers is not in the document]
Comment (jixuan1989) [reply .travis.yml]: if we test sub-modules one by one, is that ok?
maybe you can change it to:
```yaml 
script: 
   mvn clean package -Dmaven.test.skip=true
   # For each test, travis-ci limits its output log. So we cannot test some sub-modules like tsfile, iotdb.
   # We only test jdbc to ensure maven commands work correctly on travis-ci.
mvn clean test -pl tsfile -am 
mvn clean test -pl iotdb -am -Dtsfile.test.skip=true 
 mvn clean test -pl service-rpc -am -Dtsfile.test.skip=true 
mvn clean test -pl jdbc -am -Dtsfile.test.skip=true
mvn clean test -pl cli -am -Dtsfile.test.skip=true
```

**Comment by jixuan1989** [comment]
#552  is not fixed.

**Pull request merged**